uploadNeuroDB/NeuroDB/ImagingUpload.pm

```diff
@@ -247,55 +247,50 @@ sub IsCandidateInfoValid {
-                if ( !$this->isDicom($_) ) {
-                    $files_not_dicom++;
-                }
-    	        else {
             #######################################################
             #Validate the Patient-Name, only if it's not a phantom#
             ############## and the file is of type DICOM###########
             #######################################################
-                    if ($row[4] eq 'N') {
-                        # make sure the regex used for PatientNameMatch starts
-                        # with $this->{'pname'}
-                        if (!$this->PatientNameMatch($_, "^$this->{'pname'}")) {
-                            $files_with_unmatched_patient_name++;
-                        }
-                    } elsif ($row[4] eq 'Y') {
-                        # make sure the regex used for PatientNameMatch
-                        # includes "phantom" string
-                        my $lego_phantom_regex = NeuroDB::DBI::getConfigSetting(
-                            $this->{dbhr}, 'LegoPhantomRegex'
-                        );
-                        my $living_phantom_regex = NeuroDB::DBI::getConfigSetting(
-                            $this->{dbhr}, 'LivingPhantomRegex'
-                        );
-                        my $phantom_regex =
-                            "($lego_phantom_regex)|($living_phantom_regex)";
-                        if (!$this->PatientNameMatch($_, $phantom_regex)) {
-                            $files_with_unmatched_patient_name++;
-                        }
-                    }
+            if ($row[4] eq 'N') {
+                # make sure the regex used for PatientNameMatch starts
+                # with $this->{'pname'}
+                if (!$this->PatientNameMatch($_, "^$this->{'pname'}")) {
+                   $files_with_unmatched_patient_name++;
+                }
+            } elsif ($row[4] eq 'Y') {
+                # make sure the regex used for PatientNameMatch
+                # includes "phantom" string
+                my $lego_phantom_regex = NeuroDB::DBI::getConfigSetting(
+                    $this->{dbhr}, 'LegoPhantomRegex'
+                );
+                my $living_phantom_regex = NeuroDB::DBI::getConfigSetting(
+                    $this->{dbhr}, 'LivingPhantomRegex'
+                );
+                my $phantom_regex =
+                    "($lego_phantom_regex)|($living_phantom_regex)";
+                if (!$this->PatientNameMatch($_, $phantom_regex)) {
+                    $files_with_unmatched_patient_name++;
                 }
             }
         }
     }
 
     if ( $files_not_dicom > 0 ) {
-        $message = "\nERROR: There are $files_not_dicom file(s) which"
-          . " are not of type DICOM \n";
-        $this->spool($message, 'Y', $notify_notsummary);
-        return 0;
+        $message = "\nWARNING: There are $files_not_dicom file(s) which"
+          . " are not of type DICOM: these will be ignored\n";
```

Comment:
Or maybe in the warning message, we could show the list of files that were ignored or removed from the archive so at least you can keep a trace of what was removed?

Reply:
I am worried that you mind end up erasing DICOM files that are not of type DICOM or files that people might still want to keep in the archive for some reason. 

I would have a tendency to only execute the command for files that starts with `.`. What do you think?

I don't have a strong opinion about this though. Just thought I would raise this since it crossed my mind.

Reply:
I agree with Cecile here. I would be very useful to have the list of files that were ignore or removed, it can help identify files that should have not been uploaded